inlanecruising_plugin/include/inlanecruising_plugin/inlanecruising_plugin.h

```diff
@@ -234,6 +233,51 @@ class InLaneCruisingPlugin
   std::vector<PointSpeedPair> attach_back_points(const std::vector<PointSpeedPair>& points, const int nearest_pt_index, 
                                std::vector<inlanecruising_plugin::PointSpeedPair> future_points, double back_distance) const;
   
+  /**
+   * \brief outputs the yield service
+   * 
+   * \param got_service flag to show if service is received before
+   *
+   * \return yield service
+   */
+  cav_srvs::PlanTrajectory get_yield_service(bool got_service);
+
+    /**
+   * \brief set the yield service
+   * 
+   * \param yield_srv input yield service
+
+   */
+  void set_yield_service(const cav_srvs::PlanTrajectory& yield_srv);
+
+    /**
+   * \brief set the final yield trajectory plan
+   * 
+   * \param yield_plan input yield plan
+
+   */
+  void set_yield_trajectory(const cav_msgs::TrajectoryPlan& yield_plan);
+
+  /**
+   * \brief outputs the yield trajectory (helper for unit test)
+   * 
+   * \param got_trajectory flag to show if trajectory is received before
+   *
+   * \return yield trajectory plan
+   */
+  cav_msgs::TrajectoryPlan get_yield_trajectory(bool got_trajectory);
+
+    /**
+   * \brief verify if the input yield trajectory plan is valid
+   * 
+   * \param yield_plan input yield trajectory plan
+   *
+   * \return true or falss
+   */
+  bool validate_yield_plan(const cav_msgs::TrajectoryPlan& yield_plan);
+
+  bool yield_called_ = false;
```

Comment:
Mutable public variables always concern me, is there a reason this needs to be public?